src/harness/testcases/WINNF_FT_S_SSS_testcase.py

```diff
@@ -124,16 +160,121 @@ def test_WINNF_FT_S_SSS_12(self, config_filename):
 
   @winnforum_testcase
   def test_WINNF_FT_S_SSS_13(self):
-    """ Disallowed TLS method attempted during registration.
+    """ Disallowed TLS method attempted by SAS Test Harness.
 
     Checks that SAS UUT response with fatal alert message.
     """
     self.assertTlsHandshakeFailure(SAS_CERT, SAS_KEY, ssl_method=SSL.TLSv1_1_METHOD)
 
   @winnforum_testcase
   def test_WINNF_FT_S_SSS_14(self):
-    """Invalid ciphersuite presented during registration.
+    """Invalid ciphersuite presented by SAS Test Harness.
 
     Checks that SAS UUT response with fatal alert message.
     """
     self.assertTlsHandshakeFailure(SAS_CERT, SAS_KEY, ciphers='ECDHE-RSA-AES256-GCM-SHA384')
+
+  def generate_SSS_15_default_config(self, filename):
+    """Generates the WinnForum configuration for SSS_15. """
+    # Create the actual config for sas cert/key path
+
+    config = {
+        'sasCert': self.getCertFilename("sas_inapplicable.cert"),
+        'sasKey': self.getCertFilename("sas.key")
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_SSS_15_default_config)
+  def test_WINNF_FT_S_SSS_15(self, config_filename):
+    """Certificate with inapplicable fields presented during Full Activity Dump 
+    """
+
+    config = loadConfig(config_filename)
+    
+    # Notify SAS UUT about peer SAS
+    certificate_hash = getCertificateFingerprint(config['sasCert'])
+    self._sas_admin.InjectPeerSas({'certificateHash': certificate_hash,\
+                                     'url': SAS_TH_URL })
+    self.assertTlsHandshakeSucceed(self._sas_admin._base_url, ['AES128-GCM-SHA256'], SAS_CERT, SAS_KEY)
+    # Initiate Full Activity Dump
+    try:
+       response = self.TriggerFullActivityDumpAndWaitUntilComplete(config['sasCert'], config['sasKey'])
+    except AssertionError as e:
+       # Check if HTTP status is 403
+       self.assertEqual(e.args[0], 403)
+
+  def generate_SSS_17_default_config(self, filename):
+    """Generates the WinnForum configuration for SSS_17"""
+    # Create the actual config for sas cert/key path
+
+    config = {
+        'sasCert': self.getCertFilename("sas.cert"),
+        'sasKey': self.getCertFilename("sas.key")
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_SSS_17_default_config)
+  def test_WINNF_FT_S_SSS_17(self, config_filename):
+    """Unknown SAS attempts to establish a TLS session.
+    """
+    config = loadConfig(config_filename)
+    # Attempt TLS handshake without injecting peer SAS info and verify it fails
+    self.assertTlsHandshakeFailure(config['sasCert'], config['sasKey'])
```

Comment:
Hum not sure to understand how it is supposed to work. But I admit I didn't follow the related discussion on the forum so perhaps I'am completely wrong...

On the test specification we have some low level TLS handshake tests that are supposed to use only the generic openSSL check.
Then we have some specific application check managed by the CERT_ERROR.

Here we avoid to 'configure' the application level and hope that the low level TLS handshake will fail. Should the test must expect a CERT_ERROR rather than the TLS fatal error?

Reply:
Norbert, we are in process of taking care of all other review comments. For this one we have a question to you.

Currently this test case only check if the TLS handshake fails. It does not do an application level check i.e., it does not verify the HTTP error code in case the TLS succeeds (similar to SSS.15). I think you want it to be done this way. However the test spec has not mentioned it this way. 

I do not think there were any discussion on WG4 regarding this. So we suggest to stay with what the spec says. What do you think ?